scripts/buildhost/start_ec2_instance.sh

```diff
@@ -93,15 +114,21 @@ chmod 400 key.pem
 echo "Waiting for SSH connection"
 end=$((SECONDS+90))
 while [ $SECONDS -lt $end ]; do
-    ssh -i key.pem -o "StrictHostKeyChecking no" ubuntu@$(cat instance) "uname"
-    if [ "$?" = "0" ]; then
+    if ssh -i key.pem -o "StrictHostKeyChecking no" ubuntu@"$(cat instance)" "uname" ; then
         echo "SSH connection ready"
         exit 0
     fi
     sleep 1s
 done
 
 echo "error: Unable to establish SSH connection"
+
+# Fallthrough error condition - delete security group, keypair, and instance
+aws ec2 delete-security-group --group-id "$(cat sgid)"
+aws ec2 delete-key-pair --key-name "$(cat key-name)"
+aws ec2-terminate-instances --instance-ids "$(cat instance-id)"
```

Comment:
In case of an error code in any of these, we might want to keep the respective files, so that the stop_ec2_instance.sh would have a chance of trying to destroy these resources "slowly".

Reply:
As Tsachi noted we could keep the files maybe to clean up later, but can do that in a future PR. Because I'm switching to calling the shutdown script it would involve parameterizing that, but I think the build host might clean up the temp dirs anyway.